File: AndroidStudioProjects/basitverisaklama/app/src/main/java/com/example/basitverisaklama/MainActivity.java
package com.example.basitverisaklama;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    SharedPreferences saklacakVeri;
    EditText number;
    TextView ekran;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ekran = findViewById(R.id.textView);
        number =  findViewById(R.id.editText);



       saklacakVeri = this.getSharedPreferences("com.example.basitverisaklama", Context.MODE_PRIVATE);

       int b = saklacakVeri.getInt("kayd",0);
        //EĞER KAYD EDİLİN VERİ VARSA ÖNCENDEN ONU ÇAĞIRIR YOKSA defValue değeri kullanılır;
        ekran.setText(String.valueOf(b));
    }

    public void save(View view)
    {
        int a = Integer.parseInt(number.getText().toString());
        ekran.setText(String.valueOf(a));

        saklacakVeri.edit().putInt("kayd",a).apply();//kaydetme yöntemi
        ///"kayd bir anahtardır
    }
}

File: AndroidStudioProjects/BasitVeriSaklamakotlin/app/src/main/java/com/example/basitverisaklamakotlin/MainActivity.kt
package com.example.basitverisaklamakotlin

import android.content.Context
import android.content.SharedPreferences
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity()
{
    lateinit var kayitliVeri :SharedPreferences
    //lateinit ilerde tanımlanacak

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        kayitliVeri = this.getSharedPreferences("com.example.basitverisaklamakotlin",
            Context.MODE_PRIVATE)
        var a = kayitliVeri.getString("name","")
        result.text = "Kayıylı veri: " + a
    }

    fun save(v : View)
    {

        if (inputData.text.toString() != "")
        {
            var veri = inputData.text.toString()
            result.text = "Kayıylı veri: ${veri}"
            kayitliVeri.edit().putString("name",veri).apply()//kayd
        }
        else
        {
            result.text = "Kayıylı olacak veri yazınız"
        }

    }

    fun sil(v : View)
    {
        var a = kayitliVeri.getString("name","")
        if (a != "")
        {
            kayitliVeri.edit().remove("name").apply()//kaldır = remove
            result.text = "Kayıtlı veri yok"
        }
    }
}